### DISCUSSION TITLE
on: push unless it's a merge commit

### DISCUSSION THREAD

**Discussion by OfekShochat** [community/community#14204] [Actions]
Id still like my tests to be executed when a repo is prd, but there's no need to do it after but I still want to be able to test on push. I originally thought of `if: github.event.head_commit.parents` but I'm not sure. Can this be added as a filter or something?
Thanks aot

**Comment by ethomson** [comment]
Hi @OfekShochat - at the moment you cannot control this information in the workflow's trigger. You'll indeed need to do something either as a conditional in the workflow - like what you've described - or in a job step - whether that's looking at the git history or the github context. I'd like to make triggers more expressive in the future, and we'll consider this feedback if we do that. Thanks for the feedback.

**Reply by OfekShochat** [reply]
Cool, if I add == 1 to the conditional I described, will it work? And should I mark this as answer?
